### ISSUE TITLE
How to get correct post split stock prices 

### ISSUE THREAD

**Issue by sosonis** [afadil/wealthfolio#163]
How is it possible to get correct prices for stocks that had a recent split (like $SCHD or $LRCX)?

The app still does not display post split adjusted prices, even if I request a manual update. Eg below SCHD:
<img width="853" alt="Screenshot 2024-11-09 at 11 31 09" src="https://github.com/user-attachments/assets/687e088e-1336-4983-a662-618c28bc142d">

**Comment by vadikcoma** [comment]
I'm looking into wealthfolio and stock split is the only question holding me back. ChatGPT claims that wealthfolio supports stock splits, however I don't see any other references anywhere. My goal is to enter my original buys from years ago and make sure app can figure out when splits happened and adjust portfolio accordingly

**Comment by sosonis** [comment]
In recent versions the ability to manually enter a stock split has been added. However, since the historic stock price is not correct -not split adjusted in other words- (it is for NVDA though), you will get false performance and stock valuation statistics.

**Comment by vadikcoma** [comment]
In my case TSLA was calculated properly, however NVDA wasn't

**Comment by sosonis** [comment]
For Tesla I presume because the last split occurred a while ago (in 2022), so data source has been corrected. For NVDA it's strange since I get the split adjusted price correctly (but not on the slightly more recent splits of LRCX, SCHD (for eg).

This is what I am trying to understand. How the historic data mechanism works on the app.

@afadil mentioned that you can manually update the historic data to get the correct stock prices but I cannot see how to do it. Forcing a manual update on the stock has not difference. LRCX for eg still reports 90% drop in past 3 months (due to the 10:1 split).

**Comment by afadil** [comment]
Added trigger to synch quote history for a symbol in v1.0.21
![refresh-quotes](https://github.com/user-attachments/assets/4fbb288b-b844-45e2-9418-dca548ce502e)
Can you please test if this fixes the issue

**Comment by vadikcoma** [comment]
NVDA is still not fixed.

Another issue - moving holding from one portfolio to another doesn't remove value. I removed few stocks to another portfolio, however my totals in original portfolio are still as if I would have stocks there. Recalculating stocks don't affect anything

**Comment by sosonis** [comment]
With the new versions the stock prices have been updated correctly (at least the 2 ones in my portfolio that did not show correctly post split). The total historic valuation though of the portfolio is still wrong (in the past as it calculates (I guess) the new, post stock split shares quantity multiplied by the past (pre-split) price per share.

**Comment by afadil** [comment]
@sosonis try trigger portfolio update from the main page (Hover on value)

**Comment by sosonis** [comment]
I have done this multiple times but with no effect (same as the per stock manual updates before the fix of v. 1.0.21)

**Comment by afadil** [comment]
> I have done this multiple times but with no effect (same as the per stock manual updates before the fix of v. 1.0.21)

Ok I’ll have a look at this. Meanwhile You can try change your base currency in settings and then put it back. This will trigger the history calculation

**Comment by vadikcoma** [comment]
Yes, changing base currency did force recalc and it fixed my issue

**Comment by sosonis** [comment]
For me too!

**Comment by afadil** [comment]
Settings -> Market Data -> Refetch All should update the prices.
